optaweb-vehicle-routing-docs/src/main/asciidoc/development-guide.adoc

```diff
@@ -0,0 +1,253 @@
+
+For example, you may use:
+
+- `logging.level.org.optaweb.vehiclerouting=debug` to enable debug level for the backend code,
+- `logging.level.org.optaplanner.core=warn` to reduce OptaPlanner logging,
+- `logging.level.org.springframework.web.socket=trace` to access more details when investigating problems with WebSocket connection.
+
+[[frontend]]
+== Frontend
+
+////
+- PatternFly, Leaflet
+- Npm, React, Redux, TypeScript, ESLint, Cypress, `ncu`
+- Chrome, plugins
+- Docker
+////
+
+The frontend project was bootstrapped with https://create-react-app.dev/[Create React App].
+Create React App provides a number of scripts and dependencies that help with development and with building the application for production.
+
+=== Setting up development environment
+
+.Procedure
+. On Fedora, install npm by running:
++
+[source,shell]
+----
+sudo dnf install npm
+----
+
+See https://docs.npmjs.com/downloading-and-installing-node-js-and-npm[Downloading and installing Node.js and npm] for more information about installing npm.
+
+=== Install npm dependencies
+
+Unlike Maven, the npm package manager installs dependencies in `node_modules` under the project directory and does that only when requested by running `npm install`.
+Whenever the dependencies listed in `package.json` change (for example when you pull changes to the master branch) you must run `npm install` before you run the development server.
+
+.Procedure
+. Change directory to the frontend module
++
+[source,shell]
+----
+cd optaweb-vehicle-routing-frontend
+----
+
+. Install dependencies:
++
+[source,shell]
+----
+npm install
+----
+
+=== Running development server
+
+.Prerequisites
+- npm is installed.
+- npm dependencies are installed.
+
+.Procedure
+. Run development server:
++
+[source,shell]
+----
+npm start
+----
+
+. Open http://localhost:3000/ in a web browser.
+By default, the `npm start` command attempts to open this URL in your default browser.
+
+[TIP]
+.Prevent `npm start` from launching your default browser
+====
+If you don't want `npm start` to open a new browser tab each time you run it, export an environment variable `BROWSER=none`.
+
+You can use `.env.local` file to make this preference permanent.
+To do that, run:
+
+[source,shell]
+----
+echo BROWSER=none >> .env.local
+----
+====
+
+The browser will refresh the page make changes in the frontend source code.
+The development server process running in the terminal will pick up the changes as well and will print compilation and lint errors to the console.
+
+=== Running tests
+
+.Procedure
+. Run `npm test`.
+
+=== Changing backend location
+
+Use an environment variable called `REACT_APP_BACKEND_URL` to change backend URL when running `npm start` or `npm run build`.
+For example:
+
+[literal]
+....
+REACT_APP_BACKEND_URL=http://10.0.0.123:8081
+....
+
+Note that environment variables will be "baked" inside the JavaScript bundle during npm build, so you'll need to know backend location before building and deploying frontend.
```

Comment:
during the npm (code) build, so you need to know the back end location before you build and deploy the front end.